AK/StdLibExtraDetails.h

```diff
@@ -443,8 +462,62 @@ auto declval() -> T;
 
 template<typename T, typename... Args>
 inline constexpr bool IsCallableWithArguments = requires(T t) { t(declval<Args>()...); };
+
+template<typename To>
+To __test_implicit_conversion(To);
+
+template<typename From, typename To>
+inline constexpr bool IsConvertible = requires { __test_implicit_conversion<To>(declval<From>()); };
```

Comment:
For a more compact (read: no external decls) solution, perhaps this could be helpful:
```suggestion
inline constexpr bool IsConvertible = requires { declval<void(*)(To)>()(declval<From>()); };
```